py/packages/genkit/src/genkit/ai/document.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2025 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Any
+
+from genkit.ai.embedding import EmbeddingModel
+from pydantic import BaseModel
+
+
+class EmptyPart(BaseModel):
+    """Empty part of a document."""
+
+    text: None = None
+    media: None = None
+
+
+class TextPart(BaseModel):
+    """Text part of a document.
+
+    Attributes:
+        text: The string contents of the document
+    """
+
+    text: str
+    media: None = None
+
+
+class MediaPartModel(BaseModel):
+    """Object model representing media parts.
+
+    Attributes:
+        content_type: The media content type. Inferred from data uri if not
+        provided.
+        url: A `data:` or `https:` uri containing the media content.
+    """
+
+    content_type: str | None = None
+    url: str
+
+
+class MediaPart(BaseModel):
+    """Media part of a document.
+
+    Attributes:
+        media: The media contents of the document
+    """
+
+    text: None = None
+    media: MediaPartModel
+
+
+# Part type
+type Part = TextPart | MediaPart
+
+
+class DocumentData(BaseModel):
```

Comment:
there's already DocumentData in https://github.com/firebase/genkit/blob/15d507e5c78f1db820f7fc11b1bc6d7951a10fbe/py/packages/genkit/src/genkit/core/typing.py#L615

Reply:
Updated.